src/amo/components/CollectionAddAddon/index.js

```diff
@@ -0,0 +1,187 @@
+import type { AppState } from 'amo/store';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+
+import './styles.scss';
+
+export const MESSAGE_RESET_TIME = 5000;
+const MESSAGE_FADEOUT_TIME = 450;
+
+type UIStateType = {|
+  addonWasAdded: boolean,
+|};
+
+const initialUIState: UIStateType = {
+  addonWasAdded: false,
+};
+
+export type Props = {|
+  collection: CollectionType | null,
+  filters: CollectionFilters,
+|};
+
+type InternalProps = {|
+  ...Props,
+  clearTimeout: Function,
+  currentUsername: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  hasAddonBeenAdded: boolean,
+  i18n: I18nType,
+  setTimeout: Function,
+  setUIState: (state: $Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|};
+
+export class CollectionAddAddonBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    setTimeout:
+      typeof window !== 'undefined' ? window.setTimeout.bind(window) : () => {},
+    clearTimeout:
+      typeof window !== 'undefined'
+        ? window.clearTimeout.bind(window)
+        : () => {},
+  };
+
+  componentWillReceiveProps(props: InternalProps) {
+    const { hasAddonBeenAdded: hasAddonBeenAddedNew } = props;
+    const { hasAddonBeenAdded } = this.props;
+    if (hasAddonBeenAdded !== hasAddonBeenAddedNew) {
+      this.props.setUIState({
+        addonWasAdded: props.hasAddonBeenAdded,
+      });
+    }
+
+    if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
+      this.timeout = this.props.setTimeout(
+        this.resetMessageStatus,
+        MESSAGE_RESET_TIME,
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.timeout) {
+      this.props.clearTimeout(this.timeout);
+    }
+  }
+
+  onAddonSelected = (suggestion: SuggestionType) => {
+    const {
+      collection,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      filters,
+    } = this.props;
+    const { addonId } = suggestion;
+
+    invariant(addonId, 'addonId is required');
+    invariant(collection, 'collection is required');
+    invariant(currentUsername, 'currentUsername is required');
+
+    dispatch(
+      addAddonToCollection({
+        addonId,
+        collectionId: collection.id,
+        editing: true,
+        errorHandlerId: errorHandler.id,
+        filters,
+        slug: collection.slug,
+        username: currentUsername,
+      }),
+    );
+    this.props.setUIState({ addonWasAdded: false });
+  };
+
+  timeout: TimeoutID;
+
+  resetMessageStatus = () => {
+    this.props.setUIState({
+      addonWasAdded: false,
+    });
+  };
+
+  render() {
+    const { errorHandler, i18n, uiState } = this.props;
+
+    return (
+      <div className="CollectionAddAddon">
+        {errorHandler.hasError() ? (
+          errorHandler.renderError()
+        ) : (
+          <TransitionGroup className="NoticePlaceholder">
```

Comment:
Whoops, I didn't notice this last time. This class `NoticePlaceholder` needs to be prefixed with the component name (`CollectionAddAddon`). This is a convention we follow as an attempt to prevent CSS class name collisions. 

The [CSS style guide](https://github.com/mozilla/addons-frontend/blob/master/docs/css.md) explains this convention but we also need to make docs like this easier to find (see https://github.com/mozilla/addons-frontend/issues/2223).